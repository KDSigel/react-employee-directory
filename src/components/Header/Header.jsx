export default function Header() {
    return (
        <div>
            <span>Acme Inc name</span><span>A “Sign In” button if the user is signed out</span>
            <span>A “You are signed in as "email” message with a “Sign Out” button if the user is signed in</span>
        </div>
    )
}
