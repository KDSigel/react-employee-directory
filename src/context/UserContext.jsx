import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
    const userCtx = useContext(UserContext)
    if (userCtx === undefined) {
        throw new Error('useUser must be used within a UserProvider');
    }
    return userCtx;
};

export { UserProvider, useUser };
