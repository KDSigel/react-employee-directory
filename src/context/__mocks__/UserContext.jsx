import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ children, mockUser }) => {

  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
    const userCtx = useContext(UserContext)
    if (userCtx === undefined) {
        throw new Error('useUser must be used within a UserProvider');
    }
    return userCtx;
};

export { UserProvider, useUser };
