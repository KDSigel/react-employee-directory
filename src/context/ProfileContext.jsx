import { createContext, useContext, useState } from 'react';
import { getProfile } from '../services/profiles';
import { useUser } from './UserContext';


const ProfileContext = createContext();


const ProfileProvider = ({ children }) => {
  const {user} = useUser();
  const currentProfile = getProfile();
  const [profile, setProfile] = useState(
    currentProfile.name
      ? {
          name: currentProfile.name,
          email: user.email,
          bio: currentProfile.bio,
          birthday: currentProfile.birthday,
        }
      : {}
  );

  return (
    <ProfileContext.Provider value={{ profile, setProfile }}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
    const profileCtx = useContext(ProfileContext)
    if (profileCtx === undefined) {
        throw new Error('useProfile must be used within a ProfileProvider');
    }
    return profileCtx
};

export { ProfileProvider, useProfile };
