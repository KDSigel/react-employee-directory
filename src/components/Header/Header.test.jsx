import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import Header from './Header';

jest.mock('../../context/UserContext')
jest.mock('../../services/users')

it('should render the Header component', () => {
    const { container } = render(
        <UserProvider>
        <MemoryRouter> 
          <Header />
        </MemoryRouter>
        </UserProvider>
      );

      expect(container).toMatchSnapshot();
    });