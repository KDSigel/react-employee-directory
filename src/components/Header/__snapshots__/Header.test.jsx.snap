// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Header component 1`] = `
<div>
  <div>
    <span>
      Karl Inc
    </span>
    <span>
      ____________
    </span>
    <span>
      <span>
        Not logged in
        <a
          href="/login"
        >
          <button>
            Sign In
          </button>
        </a>
      </span>
    </span>
  </div>
</div>
`;
